File: flyweight/exercise/TextElementFactory.java
package flyweight.exercise;

import java.util.HashMap;
import java.util.Map;

public class TextElementFactory {
    private Map<String, TextElement> textElementCache;

    public TextElementFactory() {
        textElementCache = new HashMap<>();
    }

    public TextElement getTextElement(char character, Font font, Color color) {
        String key = character + "_" + font.toString() + "_" + color.toString();
        if (!textElementCache.containsKey(key)) {
            textElementCache.put(key, new Character(character, font, color));
        }
        return textElementCache.get(key);
    }
}

File: C#/Flyweight/Exercise/TextElementFactory.cs
using System.Security.Cryptography.X509Certificates;

namespace Flyweight.Exercise;

public sealed class TextElementFactory
{
    private readonly Dictionary<string, ITextElement> _textElementCache;

    public TextElementFactory()
        => _textElementCache = [];

    public ITextElement GetTextElement(char character, Font fontName, Color colorName)
    {
        string key = $"{character}_{fontName}_{colorName}";

        if (!_textElementCache.TryGetValue(key, out ITextElement? value))
        {
            _textElementCache.Add(key, new Character(character, fontName, colorName));
        }

        return value;
    }
}
